src/DynamoCoreWpf/Views/Search/LibraryView.xaml.cs

```diff
@@ -148,7 +130,26 @@ private void OnTreeViewItemPreviewMouseLeftButton(object sender, MouseButtonEven
                 {
                     // If class button was clicked, then handle, otherwise leave it.
                     e.Handled = selectedClass.IsClassButton;
-                    selectedElement.BringIntoView();
+
+                    var classInfoPanel = wrapPanel.Children.Cast<FrameworkElement>().
+                        Where(child => child.DataContext is ClassInformationViewModel).FirstOrDefault();
+
+                    classInfoPanel.Loaded += (send, handler) =>
+                        {
+                            var selectedClassButton = WpfUtilities.FindUpVisualTree<Border>(selectedElement);
+                            var height = classInfoPanel.RenderSize.Height + selectedClassButton.RenderSize.Height;
+                            if (height > ScrollLibraryViewer.ActualHeight)
+                            {
+                                selectedClassButton.BringIntoView(new Rect(0, 0, 0, ScrollLibraryViewer.ActualHeight));
+                            }
+                            else
+                            {
+                                if (IsUserVisible(selectedClassButton, ScrollLibraryViewer))
```

Comment:
You are absolutely right. That will bring just class button. But that is exactly, what we need.
That is needed, when some category with long standard panel open(e.g. Geometry). If we click item under it (e.g. Line). Geometry will collapse and Standard panel of Line will appear, but Line will be hidden.
E.g.:
![image](https://cloud.githubusercontent.com/assets/8158404/7583329/03bb034e-f89d-11e4-8878-4a830ed5f30f.png)
In image above you can see, that Line is opened but is out of scrollviewer. In this case we should bring into view class button.